go/controller/internal/utils/common.go

```diff
@@ -1,7 +1,33 @@
 package common
 
-import "os"
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"unicode"
 
+	"github.com/kagent-dev/kagent/go/controller/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// ResourceInfo represents a Kubernetes resource that can be associated with a ModelConfig.
+// It extends client.Object to provide access to standard Kubernetes object metadata
+// while adding the ability to specify which ModelConfig should be used for the resource.
+// Implementers must provide a GetModelConfigName() method that returns either:
+// - An empty string: indicating the default ModelConfig should be used
+// - A name: indicating a ModelConfig in the same namespace as the resource
+// - A namespace/name reference: indicating a specific ModelConfig in a specific namespace
+type ResourceInfo interface {
```

Comment:
Can we call this something slightly more releavnt to the impl. For example: `ObjectWithModelConfig` or something like that. `ResourceInfo` is pretty generic.